src/core/server/saved_objects/migrationsv2/integration_tests/outdated_docs.test.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import Fs from 'fs';
+import Util from 'util';
+import { kibanaPackageJson as pkg } from '@kbn/utils';
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+import type { ElasticsearchClient } from '../../../elasticsearch';
+import { Root } from '../../../root';
+
+const logFilePath = Path.join(__dirname, 'migration_test_kibana.log');
+
+const asyncUnlink = Util.promisify(Fs.unlink);
+async function removeLogFile() {
+  // ignore errors if it doesn't exist
+  await asyncUnlink(logFilePath).catch(() => void 0);
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+
+  beforeAll(async () => {
+    await removeLogFile();
+  });
+
+  afterAll(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  it('migrates the documents to the highest version', async () => {
+    const migratedIndex = `.kibana_${pkg.version}_001`;
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+          // original SO:
+          // {
+          //  type: 'foo',
+          //  foo: {},
+          //  migrationVersion: {
+          //    foo: '7.13.0',
+          //  },
+          // },
+          // contains migrated index with 8.0 aliases to skip migration, but run outdated doc search
+          dataArchive: Path.join(__dirname, 'archives', '8.0.0_migrated_with_outdated_docs.zip'),
```

Comment:
>  with 8.0 aliases

We'll probably need to disable this test in the `7.x` branch, as I think it will fail because the alias points to an higher version of Kibana?

But I don't see any other option to test such scenario on `master`.

Reply:
discussed offline: it's okay to skip the test on `7.x` branch to reduce the maintenance cost of different fixtures for 8 and 7.x branches